### DISCUSSION TITLE
Using Brew via Intune - FeedBack

### DISCUSSION THREAD

**Discussion by stephenc01** [Homebrew/discussions#3626] [Everyday usage]
I am looking for general feed back or maybe a better way to do it. Intune MacOS support leaves a lot to be desired. 

I am currently deploying brew nested in side an AppleScript App via Intune to the Macs we support. 

Every 6 hours I run this process
`export PATH="/Applications/homebrew.app/Contents:$PATH"`
`eval "$(/Applications/homebrew.app/Contents/homebrew/bin/brew shellenv)"`
`brew update --force --quiet`
`chmod -R go-w "$(brew --prefix)/share/zsh"`
`brew upgrade --quiet --cask`
`brew cleanup --quiet`

This is the part that bothers me. Since some apps require passwords to install. While the script is running I add the user to the sudoers file to run anything then remove it.

`echo "$LastUser ALL=(ALL) NOPASSWD: ALL" > /etc/sudoers.d/MacBrew `
`su - $LastUser -c "/Applications/homebrew.app/Contents/script.sh"`
`rm /etc/sudoers.d/MacBrew `

**Comment by swissbuechi** [comment]
I would suggest you to handle automstic updates via https://github.com/homebrew/homebrew-autoupdate.

The `-sudo` argument whil handle the casks that require sudo.
